ts/packages/cli/src/commands/find.ts

```diff
@@ -0,0 +1,165 @@
+      this.processLineByLine(gitIgnore, (line, lineIndex) => {
+        if (!line.startsWith('#') && !ignoredFiles.has(line)) ignoredFiles.add(line);
+      }),
+    );
+    await Promise.all(gitIgnorePromises);
+    return ignoredFiles;
+  }
+
+  private async processLineByLine(filePath: string, onLine: (line: string, lineIndex: number) => any): Promise<any[]> {
+    const readLine = createInterface({
+      input: createReadStream(filePath),
+      crlfDelay: Number.POSITIVE_INFINITY,
+    });
+    const result: any[] = [];
+    let lineIndex = 1;
+    readLine.on('line', (line) => {
+      const lineResult = onLine(line, lineIndex);
+      if (lineResult) result.push(lineResult);
+      lineIndex += 1;
+    });
+    await once(readLine, 'close');
+    return result;
+  }
+
+  public async run(): Promise<void> {
+    const findInDirectory = path.resolve(this.flags.dir);
+    const showUnusedOnly = this.flags.unused;
+    const showTemplateKeys = this.flags.templates;
+    const ignoredFiles = await this.updateIgnoredFiles(findInDirectory);
+    const cfguPaths =
+      this.flags.include.length > 0
+        ? this.flags.include
+        : FastGlob.sync([path.join(findInDirectory, '/**/*.cfgu.json')], {
+            ignore: [...ignoredFiles].map((ignoreFile) => path.join(findInDirectory, `/**/${ignoreFile}`)),
+          });
+
+    if (cfguPaths.length === 0)
+      throw new Error(
+        `no cfgu file found or provided. If you have another type of configuration file, run configu init --import=<CONFIG-FILE> --defaults to generate a cfgu from it`,
+      );
+
+    const keysFromCfgu = await Promise.all(
+      cfguPaths.map(async (cfguFile) => {
+        try {
+          const schemaContents = await ConfigSchema.parse(new ConfigSchema(cfguFile));
+          if (!showTemplateKeys) {
+            const schemaEntries = Object.entries(schemaContents);
+            const ignoredKeys = new Set(
+              schemaEntries
+                .filter(([key, cfgu]) => !!cfgu.template)
+                .flatMap(([key, cfgu]) => {
+                  return TMPL.parse(cfgu.template as string)
+                    .filter((templateSpan) => templateSpan.type === 'name')
+                    .map((templateSpan) => templateSpan.key);
+                }),
+            );
+            return schemaEntries.filter(([key, cfgu]) => !ignoredKeys.has(key)).map(([key, cfgu]) => key);
+          }
+          return Object.keys(schemaContents);
+        } catch {
+          return [];
+        }
+      }),
+    );
+
+    const keysRegEx = _.mapValues(_.keyBy([...new Set(keysFromCfgu.flat())]), (key) => ({
+      pattern: new RegExp(key, 'g'),
+      count: 0,
+    }));
+
+    const files = FastGlob.sync([path.join(findInDirectory, '/**/*.*')], {
+      dot: true,
+      ignore: [...[...ignoredFiles].map((dirName) => path.join(findInDirectory, `/**/${dirName}`)), '/**/*.json'],
+    });
+
+    await Promise.all(
+      files.map((file) =>
```

Comment:
please keep the same pattern with all of the `Promise.all` as stated in the previous comments